src/machine/usb/descriptor/hid.go

```diff
@@ -0,0 +1,202 @@
+package descriptor
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+)
+
+var configurationCDCHID = [configurationTypeLen]byte{
+	configurationTypeLen,
+	TypeConfiguration,
+	0x64, 0x00, // adjust length as needed
+	0x03, // number of interfaces
+	0x01, // configuration value
+	0x00, // index to string description
+	0xa0, // attributes
+	0x32, // maxpower
+}
+
+var ConfigurationCDCHID = ConfigurationType{
+	data: configurationCDCHID[:],
+}
+
+var interfaceHID = [interfaceTypeLen]byte{
+	interfaceTypeLen,
+	TypeInterface,
+	0x02, // InterfaceNumber
+	0x00, // AlternateSetting
+	0x01, // NumEndpoints
+	0x03, // InterfaceClass
+	0x00, // InterfaceSubClass
+	0x00, // InterfaceProtocol
+	0x00, // Interface
+}
+
+var InterfaceHID = InterfaceType{
+	data: interfaceHID[:],
+}
+
+const (
+	ClassHIDTypeLen = 9
+)
+
+type ClassHIDType struct {
+	data []byte
+}
+
+func (d ClassHIDType) Bytes() []byte {
+	return d.data[:]
+}
+
+func (d ClassHIDType) Length(v uint8) {
+	d.data[0] = byte(v)
+}
+
+func (d ClassHIDType) Type(v uint8) {
+	d.data[1] = byte(v)
+}
+
+func (d ClassHIDType) HID(v uint16) {
+	binary.LittleEndian.PutUint16(d.data[2:4], v)
+}
+
+func (d ClassHIDType) CountryCode(v uint8) {
+	d.data[4] = byte(v)
+}
+
+func (d ClassHIDType) NumDescriptors(v uint8) {
+	d.data[5] = byte(v)
+}
+
+func (d ClassHIDType) ClassType(v uint8) {
+	d.data[6] = byte(v)
+}
+
+func (d ClassHIDType) ClassLength(v uint16) {
+	binary.LittleEndian.PutUint16(d.data[7:9], v)
+}
+
+var errNoClassHIDFound = errors.New("no classHID found")
+
+// FindClassHIDType tries to find the ClassHID data in the descriptor.
+func FindClassHIDType(data, section []byte) (ClassHIDType, error) {
+	if len(data) < ClassHIDTypeLen || len(section) == 0 {
+		return ClassHIDType{}, errNoClassHIDFound
+	}
+
+	// search only for ClassHIDType without the ClassLength,
+	// in case it has already been set.
+	idx := bytes.Index(data[:ClassHIDTypeLen-2], section)
```

Comment:
ClassHIDTypeLen is 9, so this will not work.

```suggestion
	idx := bytes.Index(data, section)
```

Reply:
Oops, I had it reversed. Renamed the parameter names and corrected, seems to be working now.